File: data structures and algorithms/java/08_070.java
public class Solution {
   public int maxHolidayJoy(int[] joyRatings, 
         int minCards, int maxCards) {
      // Return 0 if ratings array is too small
      int n = joyRatings.length;
      if (n < minCards) {
         return 0;
      }
      
      // Calculate initial minimum window sum
      int currentSum = 0;
      for (int i = 0; i < minCards; i++) {
         currentSum += joyRatings[i];
      }
      int maxJoy = currentSum;
      
      // Try each valid window size
      for (int windowSize = minCards; 
           windowSize <= Math.min(maxCards, n); 
           windowSize++) {
         // Expand window if larger than minimum
         if (windowSize > minCards) {
            currentSum += joyRatings[windowSize - 1];
         }
         
         // Initial window of current size
         int tempMax = currentSum;
         
         // Slide window across array
         for (int i = windowSize; i < n; i++) {
            currentSum += joyRatings[i] - 
                  joyRatings[i - windowSize];
            tempMax = Math.max(tempMax, currentSum);
         }
         
         maxJoy = Math.max(maxJoy, tempMax);
         
         // Reset current sum for next window
         currentSum = 0;
         for (int i = 0; i < windowSize; i++) {
            currentSum += joyRatings[i];
         }
      }
      
      return maxJoy;
   }
}

File: data structures and algorithms/cs/08_070.cs
using System;

public class Solution
{
   public int MaxHolidayJoy(
      int[] joyRatings,
      int minCards,
      int maxCards)
   {
      int n = joyRatings.Length;
      if (n < minCards)
      {
         return 0;
      }

      int currentSum = 0;
      for (int i = 0; i < minCards; i++)
      {
         currentSum += joyRatings[i];
      }
      int maxJoy = currentSum;

      for (int windowSize = minCards;
         windowSize <= Math.Min(maxCards, n);
         windowSize++)
      {
         if (windowSize > minCards)
         {
            currentSum += joyRatings[windowSize - 1];
         }

         int tempMax = currentSum;

         for (int i = windowSize; i < n; i++)
         {
            currentSum += joyRatings[i] -
               joyRatings[i - windowSize];
            tempMax = Math.Max(tempMax, currentSum);
         }

         maxJoy = Math.Max(maxJoy, tempMax);

         currentSum = 0;
         for (int i = 0; i < windowSize; i++)
         {
            currentSum += joyRatings[i];
         }
      }

      return maxJoy;
   }
}
